chapters/documentation.Rmd

```diff
@@ -11,6 +11,59 @@ the best function names in the world aren't going to answer the questions
 and
 "Why doesn't it do this in a simpler way?"
 
+In this appendix we address some issues that commonly arise
+when people start to think about documenting their software.
+We start with tips for writing good docstrings,
+before considering more detailed documentation
+like tutorials, cookbooks and frequently-asked question
+(\gref{FAQ}{faq}\index{FAQ (Frequency Asked Questions)}\index{Python package!FAQ}) lists.
+Clearly defining your audience can make it easier to determine
+which types of documentation you need to provide,
+and there are a number of things you can do to reduce the
+work involved in providing support for FAQs. 
+
+## Writing Good Docstrings {#documentation-docstrings}
+
+If we are doing \gref{exploratory programming}{exploratory_programming},
+a short \gref{docstring}{docstring} to remind ourselves of each function's purpose\index{docstring (in Python)}\index{Python!docstring}
+is probably as much documentation as we need.
+(In fact, it's probably better than what most people do.)
+That one- or two-liner should begin with an active verb and describe either
+how inputs are turned into outputs,
+or what side effects the function has;
+as we discuss below,
+if we need to describe both,
+we should probably rewrite our function.
+
+An active verb is something like "extract", "normalize", or "find".
+For example,
+these are all good one-line docstrings:
```

Comment:
I like this section a lot - is it worth adding a couple of examples of *bad* docstrings for contrast?

Reply:
For each of the three good examples, perhaps we should show what a bad docstring for the same function would be?

The other glaring omission from this section is that we give a five point list of things to include in docstrings (name and purpose of every parameter, etc), but we don't give an example of what a docstring with all that information in it looks like, nor do we talk about the option of using reStructuredText, Google or Numpydoc style (e.g. [this post](http://daouzli.com/blog/docstring.html)) to format that information.

Reply:
I might merge this PR as is and start a new for the suggested docstring content...